packages/gatsby-plugin-utils/src/validate.ts

```diff
@@ -28,3 +32,147 @@ export async function validateOptionsSchema(
 
   return value
 }
+
+async function validatePluginsOptions(
+  plugins: Array<IPluginRefObject>,
+  rootDir: string | null
+): Promise<{
+  errors: number
+  plugins: Array<IPluginRefObject>
+}> {
+  let errors = 0
+  const newPlugins = await Promise.all(
+    plugins.map(async plugin => {
+      let gatsbyNode
+
+      try {
+        gatsbyNode = require(`${plugin.resolve}/gatsby-node`)
+      } catch (err) {
+        gatsbyNode = {}
+      }
+
+      if (!gatsbyNode.pluginOptionsSchema) return plugin
+
+      let optionsSchema = (gatsbyNode.pluginOptionsSchema as Exclude<
+        GatsbyNode["pluginOptionsSchema"],
+        undefined
+      >)({
+        Joi,
+      })
+
+      // Validate correct usage of pluginOptionsSchema
+      if (!Joi.isSchema(optionsSchema) || optionsSchema.type !== `object`) {
+        console.warn(
```

Comment:
ℹ️ I've switched from reporter to console. I didn't really know how to get a reporter here :/

Reply:
What I've done in some places is allow the calling function to pass-in the reporter, but if they don't, fall-back to a shim like the one in create-gatsby